core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/CaseExpression.kt

```diff
@@ -4,156 +4,108 @@ import ch.ergon.dope.DopeQuery
 import ch.ergon.dope.resolvable.Resolvable
 import ch.ergon.dope.resolvable.expression.TypeExpression
-import ch.ergon.dope.resolvable.expression.UnaliasedExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.function.conditional.SearchResult
 import ch.ergon.dope.validtype.BooleanType
 import ch.ergon.dope.validtype.ValidType
 
 private const val CASE = "CASE"
 private const val WHEN = "WHEN"
 private const val THEN = "THEN"
-private const val END = "END"
 private const val ELSE = "ELSE"
+private const val END = "END"
 
-class SimpleCaseClass<T : ValidType>(private val case: TypeExpression<T>) : Resolvable {
+class CaseClass<T : ValidType>(private val case: TypeExpression<T>? = null) : Resolvable {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val caseDopeQuery = case.toDopeQuery(manager)
-        return DopeQuery("$CASE ${caseDopeQuery.queryString}", caseDopeQuery.parameters)
+        val caseDopeQuery = case?.toDopeQuery(manager)
+        return DopeQuery("$CASE${caseDopeQuery?.queryString?.let { " $it" }.orEmpty()}", caseDopeQuery?.parameters.orEmpty())
     }
 }
 
-class SimpleCaseExpression<T : ValidType, U : ValidType>(
-    val case: SimpleCaseClass<T>,
+open class CaseExpression<T : ValidType, U : ValidType>(
+    val case: CaseClass<T>,
     val firstSearchResult: SearchResult<T, out U>,
-    vararg val additionalSearchResult: SearchResult<T, out U>,
+    open vararg val additionalSearchResult: SearchResult<T, out U>,
 ) : TypeExpression<U> {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
         val caseDopeQuery = case.toDopeQuery(manager)
-        val whenThenDopeQueries = listOf(firstSearchResult, *additionalSearchResult).map {
+        val conditionDopeQueries = listOf(firstSearchResult, *additionalSearchResult).map {
             it.searchExpression.toDopeQuery(manager) to it.resultExpression.toDopeQuery(manager)
         }
         return DopeQuery(
             queryString = caseDopeQuery.queryString +
-                whenThenDopeQueries.joinToString(" ", " ", " ") { "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}" } +
+                conditionDopeQueries.joinToString(" ", " ", " ") { "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}" } +
                 END,
-            parameters = caseDopeQuery.parameters + whenThenDopeQueries.fold(emptyMap()) { parameters, query ->
+            parameters = caseDopeQuery.parameters + conditionDopeQueries.fold(emptyMap()) { parameters, query ->
                 parameters + query.first.parameters + query.second.parameters
             },
         )
     }
 }
 
-class SimpleElseCaseExpression<T : ValidType, U : ValidType>(
-    private val case: SimpleCaseClass<T>,
+open class ElseCaseExpression<T : ValidType, U : ValidType>(
+    private val case: CaseClass<T>,
     private val firstSearchResult: SearchResult<T, out U>,
     private vararg val additionalSearchResult: SearchResult<T, out U>,
```

Comment:
Do you need all these three, can't you just have a reference to a CaseExpression and call the toDopeQuery from that?

Reply:
Sadly not because of the "END" in the query String